.buildkite/bisect/bisect.rayci.yml

```diff
@@ -5,9 +5,19 @@ steps:
   - name: macos test
     commands:
       - if [[ "$(buildkite-agent meta-data get test-type)" != "macos_test" ]]; then exit 0; fi
-      - RAYCI_BISECT_RUN=1 ./ci/ray_ci/macos/macos_ci.sh bisect "$(buildkite-agent meta-data get test-name)" 
+      - RAYCI_BISECT_RUN=1 ./ci/ray_ci/macos/macos_ci.sh bisect 
+          "$(buildkite-agent meta-data get test-name)" "$(buildkite-agent meta-data get test-type)"
           "$(buildkite-agent meta-data get passing-commit)" "$(buildkite-agent meta-data get failing-commit)"
     mount_buildkite_agent: true
     job_env: MACOS
     instance_type: macos
     priority: 10
+
+  - name: linux test
+    commands:
+      - if [[ "$(buildkite-agent meta-data get test-type)" != "linux_test" ]]; then exit 0; fi
```

Comment:
why we are not using env var for `test-type`? if we do that, we can actually use `if` in the step to skip the step entirely?

Reply:
yes that was the original design (https://github.com/ray-project/ray/pull/44786/files) but that requires replacement of the following env in the rayci yaml config so you were suggesting to use buildkite-agent instead

```
RAYCI_BISECT_TEST_NAME: __RAYCI_BISECT_TEST_NAME__
RAYCI_BISECT_PASSING_COMMIT: __RAYCI_BISECT_PASSING_COMMIT__
RAYCI_BISECT_FAILING_COMMIT: __RAYCI_BISECT_FAILING_COMMIT__
```

Reply:
that was for the mac test? because you are trying to upload sub pipeline definitions with dynamic env var values.

this one however, it is creating a new buildkite build with buildkite api, which can specify env vars on the build at the beginning? and hence `if` condition will just work out of the box without any change to `rayci`?

like if mac tests were to be triggered the same way, you could also use env vars? as it is just some test jobs on the pipeline, but not specifically related to if it is a mac test or linux test?

Reply:
- first of all. I will respect your decision on this.
- I am just saying that if it uses env var, it can avoid the need to have a job or using an ec2 instance there but just to run nothing.
- fwiw, human can also specify env vars? or learn to specify env vars? although having a form is probably easier to use as it is more GUI-ish.
- and we can support both even.. or maybe macos bisect can be on a separate pipeline as it is special?

either way, I think it is a small thing. I am just trying to point out the design options and results.